### PR TITLE
api: adds api authentication for user request

### PR DESCRIPTION
**Pull request by jstar1**
This is not complete, but I want to see if I am going
in the right direction.

### CODE REVIEW

**Comment by jstar1** [comment]
Hey Andy can you take a look at this and give me some feedback?

Thank you

File: routes/api.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+var express = require('express');
+var router = express.Router();
+
+var bodyParser = require('body-parser');
+
+router.post('/api/user-access', function(req, res) {
+    var stationState = 'Enabled';
+    
+    if(!req.body || req.body.length === 0 || req.body > 200) {
```
Comment (mongolsamurai) [inline routes/api.js]: I think you want `|| req.body.length > 200`

Comment (mongolsamurai) [inline routes/api.js]: I think you want `|| req.body.length > 200`

File: routes/api.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+var express = require('express');
+var router = express.Router();
+
+var bodyParser = require('body-parser');
+
+router.post('/api/user-access', function(req, res) {
```
Comment (mongolsamurai) [inline routes/api.js]: I think you could put `router.post('/user-access', ...`
and then when you mount these route handlers in `app.js` you can have it prepend '/api' to the beginning of all of the routes in this module by doing something like this:
```
api = require('./routes/api')
app.use('/api', api)
````
it looks like this is what's being done on lines 9 and 27

File: routes/api.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    var stationState = 'Enabled';
+    
+    if(!req.body || req.body.length === 0 || req.body > 200) {
+        stationState = 'Disabled';
+        res.set({
+            'Content-Type': 'text/plain',
+            'Station-State': stationState,
+            'Status': res.status(400)
```
Comment (mongolsamurai) [inline routes/api.js]: I believe this should be moved to a separate call, something like this:
`res.status(400).end()`
I still haven't written a line of node, so I may have the syntax wrong. the main thing though is that response status isn't a header, it's its own special field in the response stream.
https://expressjs.com/en/4x/api.html#res.status

**Pull request closed without merging**